### ISSUE TITLE
Module '@segment/loosely-validate-event@^1.1.2' not found when installing via pnpm

### ISSUE THREAD

**Issue by onlywei** [segmentio/analytics-node#98]
Hey, what is this dependency that is inside a special NPM scope? I can't install this given that I am using NPM enterprise and `pnpm`. Is it possible for this project to not list a specially scoped dependency?

**Comment by realyze** [comment]
Getting that too.

**Comment by stephenmathieson** [comment]
I'm not sure what `pnpm` is or how it works, but [`@segment/loosely-validate-event`](https://www.npmjs.com/package/@segment/loosely-validate-event) is a publicly available [scoped package](https://docs.npmjs.com/misc/scope). Scoped packages have been around since npm v2, and npm just released [v5](https://github.com/npm/npm/blob/latest/CHANGELOG.md#v500-2017-05-25).

It sounds to me like `pnpm` is just missing crucial features. This seems like an issue that should be opened with `pnpm`, not this library.

**Comment by stephenmathieson** [comment]
Also, it looks like `pnpm` is basically [`Yarn`](https://github.com/yarnpkg/yarn), only without necessary features and with a much smaller following. Have you thought about using Yarn instead?

**Comment by onlywei** [comment]
Yes of course I am aware of Yarn and have evaluated it in detail (opened GitHub issues and Pull Requests) to that project. Ultimately we decided that it was not a good fit for our use case because it was missing an even more crucial feature than `pnpm`.

**Comment by stephenmathieson** [comment]
I'm waiting on a response on https://github.com/pnpm/pnpm/issues/776, as this does seem like an issue with `pnpm`, and we don't actually support package managers other than `npm` and `yarn`.

Out of curiosity, what features does `pnpm` bring that`yarn` lacks? Since switching to yarn, I've haven't had a single issue with it, nor have I felt that it's missing anything vital.

**Comment by onlywei** [comment]
@stephenmathieson I'd be happy to share my thoughts about `yarn` with you in a more private messaging, as I don't feel right sharing them in public. For my email, if you work at segment, I sent an email to friends@segment.com yesterday -- you should be able to identify me based on the contents of my request.

**Comment by stephenmathieson** [comment]
Unfortunately, it's not me checking that email account. You can always reach me at stephen.mathieson@segment.com tho.

**Comment by zkochan** [comment]
Sorry for the slow reaction, I was out of town... I have responded to the related issue in the pnpm repo.

In a nutshell, pnpm supports scoped packages. I tried and pnpm installed this package with no issues. @onlywei if you can provide me some steps to repro, then I'll try to help.

@stephenmathieson pnpm has a lot more new ideas than yarn (also it was public earlier). TBH, I don't see reasons to use yarn instead of npm5. pnpm does not work with everything currently because of its weird `node_modules` structure but its strictness helps to write better node packages. I have written an article about [Why should we use pnpm](https://www.kochan.io/nodejs/why-should-we-use-pnpm.html) and [pnpm's strictness helps to avoid silly bugs](https://www.kochan.io/nodejs/pnpms-strictness-helps-to-avoid-silly-bugs.html)

@stephenmathieson this if off topic but I'll write you an email as well, I have a question regarding segment's open fellowship.

**Comment by onlywei** [comment]
I am in fact using NPM Enterprise set to an automatic whitelist policy and not using the default npm registry. I am sure it will work if I used the default NPM registry, perhaps there are some settings I can tweak in my NPM Enterprise instance settings. I will investigate there and report back.

**Comment by MarcAMartin** [comment]
I'm currently having this issue with npm and pnpm

**Comment by vadimdemedes** [comment]
@MarcAMartin Are you not able to install `analytics-node` via both npm and pnpm? I just confirmed I could install it even with pnpm.

**Comment by vadimdemedes** [comment]
Closing this issue for now, since installation works on npm/pnpm/yarn. If it still happens, feel free to reopen this or create a new issue.
